independent-publisher-connectors/Google Photos/apiDefinition.swagger.json

```diff
@@ -0,0 +1,161 @@
+        "responses": {
+          "default": {
+            "description": "default",
+            "schema": {}
+          }
+        }
+      }
+    },
+    "/v1/mediaItems/%7B%7BmediaItemId%7D%7D": {},
+    "/v1/mediaItems/%7B%7BmediaItem.id%7D%7D": {},
+    "/v1/mediaItems:search": {},
+    "/v1/mediaItems:batchGet": {
+      "get": {
+        "summary": "Create Media Item(s) in Google photo library",
+        "description": "\nReturns the list of media items for the specified media item identifiers. Items are returned in the same order as the supplied identifiers.\n\n### HTTP request\n\n`GET https://photoslibrary.googleapis.com/v1/mediaItems:batchGet`\n\nThe URL uses [gRPC Transcoding](https://github.com/googleapis/googleapis/blob/master/google/api/http.proto) syntax.\n\n### Query parameters\n\n|Parameters|||\n|-|-|-\n|`mediaItemIds`|`string`|Identifiers of the media items to be requested. Must not contain repeated identifiers and cannot be empty. The maximum number of media items that can be retrieved in one call is 50.\n\n### Request body\n\nThe request body must be empty.\n\n### Response body\n\nIf successful, the response body contains data with the following structure:\n\nResponse to retrieve a list of media items.\n\nJSON representation\n\n```json\n{\n  \"mediaItemResults\": [\n    {\n      object (MediaItemResult)\n    }\n  ]\n}\n```\n\n|Fields|||\n|-|-|-\n|`mediaItemResults[]`|`object` ([MediaItemResult](https://developers.google.com/photos/library/reference/rest/v1/mediaItems/batchGet#MediaItemResult))|Output only. List of media items retrieved. Note that even if the call to mediaItems.batchGet succeeds, there may have been failures for some media items in the batch. These failures are indicated in each [MediaItemResult.status](https://developers.google.com/photos/library/reference/rest/v1/mediaItems/batchGet#MediaItemResult.FIELDS.status).\n\n### Authorization Scopes\n\nRequires one of the following OAuth scopes:\n\n*   `https://www.googleapis.com/auth/photoslibrary`\n*   `https://www.googleapis.com/auth/photoslibrary.readonly`\n*   `https://www.googleapis.com/auth/photoslibrary.readonly.appcreateddata`\n\nMediaItemResult\n---------------\n\nResult of retrieving a media item.\n\nJSON representation\n\n```json\n{\n  \"status\": {\n    object (Status)\n  },\n  \"mediaItem\": {\n    object (MediaItem)\n  }\n}\n```\n\n|Fields|||\n|-|-|-\n|`status`|`object` ([Status](https://developers.google.com/photos/library/reference/rest/v1/Status))|If an error occurred while accessing this media item, this field is populated with information related to the error. For details regarding this field, see [Status](https://developers.google.com/photos/library/reference/rest/v1/Status).\n|`mediaItem`|`object` ([MediaItem](https://developers.google.com/photos/library/reference/rest/v1/mediaItems#MediaItem))|Media item retrieved from the user's library. It's populated if no errors occurred and the media item was fetched successfully.\n",

```

Comment:
@juliamuiruri4 the description will show up as the operation description when selecting the information button. We suggest keeping it short to fit in the text box. No period required if there's a single word. Please see [here](https://learn.microsoft.com/en-us/connectors/custom-connectors/connector-string-guidance#operations)